clippy_utils/src/check_proc_macro.rs

```diff
@@ -244,6 +245,23 @@ fn variant_search_pat(v: &Variant<'_>) -> (Pat, Pat) {
     }
 }
 
+fn fn_kind_pat(kind: &FnKind<'_>) -> (Pat, Pat) {
+    let (start_pat, end_pat, visibility) = match kind {
+        FnKind::ItemFn(.., header, visibility) => (fn_header_search_pat(*header), Pat::Str(""), Some(visibility.node)),
+        FnKind::Method(.., sig, visibility) => (
+            fn_header_search_pat(sig.header),
+            Pat::Str(""),
+            visibility.as_ref().map(|spanned| spanned.node),
+        ),
+        FnKind::Closure => return (Pat::Str(""), Pat::Str("")),
```

Comment:
A pattern for the body should be used here. Same as is done for expressions.

Reply:
You can pass around `BodyId` instead of `Body`. It would save the map lookup until it's actually required.

Reply:
I get the `Body` for free as part of `LateLintPass::check_fn`.

Reply:
That is a good point.